### ISSUE TITLE
Parcel file watcher watches Python `lib` directory instead of the project folder when `ms-python.vscode-pylance` is enabled

### ISSUE THREAD

**Issue by sztal** [microsoft/pylance-release#5011]
Type: <b>Bug</b>

The problem as in the title. This ticket is a follow-up of https://github.com/microsoft/vscode/issues/196566 where some additional discussion can be found.

**In short:** the diagnosis of the problem as described in the title has been confirmed using VS Code troubleshooting utility and bisection-based disabling/enabling of all extensions. Moreover, it is also confirmed by logging info:

```
TRACE [File Watcher (parcel)] Request to start watching: /home/sztal/miniconda3/lib/python3.11 
(excludes: **/.git/objects/**,**/.git/subtree-cache/**,**/node_modules/*/**,**/.hg/store/**,**/__pycache__,**/.DS_Store,**/.git,**/.hg,**/.mypy_cache,**/.nox,**/.pytest_cache,**/.scrapy,**/.svn,**/.tox,**/CVS, includes: <all>)
```

Note that indeed `parcel` is watching the Python installation folder instead of the project folder (which in this case was in an entirely different place lik `/hom/sztal/Projects/...`

Below is `files.watcherExclude` part of my settings.

```
"files.watcherExclude": {
        "**/__pycache__": true,
        "**/.DS_Store": true,
        "**/.git": true,
        "**/.git/objects/**": true,
        "**/.git/subtree-cache/**": true,
        "**/.hg": true,
        "**/.mypy_cache": true,
        "**/.nox": true,
        "**/.pytest_cache": true,
        "**/.scrapy": true,
        "**/.svn": true,
        "**/.tox": true,
        "**/CVS": true,
        "**/node_modules/*/**": true
    },
```

And here is the number of allowed system watchers:
```bash
cat /proc/sys/fs/inotify/max_user_watches  # output: 65536
```

I hope that system and VS Code info attached to the ticket will be helpful.

VS Code version: Code 1.83.1 (f1b07bd25dfad64b0167beb15359ae573aecd2cc, 2023-10-10T23:45:31.402Z)
OS version: Linux x64 6.5.6-76060506-generic
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 9 5900HX with Radeon Graphics (16 x 4466)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|1, 1, 1|
|Memory (System)|30.70GB (22.44GB free)|
|Process Argv|--unity-launch --crash-reporter-id b93d1314-2224-4c26-bfac-250caede6893|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|pop|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|pop|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (43)</summary>

Extension|Author (truncated)|Version
---|---|---
preview-pdf|ana|1.0.0
github-markdown-preview|bie|0.3.0
markdown-checkbox|bie|0.4.0
markdown-emoji|bie|0.3.0
markdown-footnotes|bie|0.1.1
markdown-mermaid|bie|1.20.0
markdown-preview-github-styles|bie|2.0.3
markdown-yaml-preamble|bie|0.1.0
ndjson-colorizer|bus|1.0.0
vscode-great-icons|emm|2.1.102
file-icons|fil|1.1.0
latex-count|geo|1.2.1
gc-excelviewer|Gra|4.2.58
latex-workshop|Jam|9.14.1
json-lines-viewer|leh|0.0.4
restructuredtext|lex|189.3.0
isort|ms-|2023.10.1
mypy-type-checker|ms-|2023.4.0
pylint|ms-|2023.8.1
python|ms-|2023.18.0
vscode-pylance|ms-|2023.10.40
jupyter|ms-|2023.9.1102792234
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.17
vscode-jupyter-cell-tags|ms-|0.1.8
vscode-jupyter-slideshow|ms-|0.1.5
remote-ssh|ms-|0.106.5
remote-ssh-edit|ms-|0.86.0
cpptools|ms-|1.17.5
makefile-tools|ms-|0.7.0
remote-explorer|ms-|0.4.1
material-icon-theme|PKi|4.31.0
quarto|qua|1.105.0
vscode-yaml|red|1.14.0
rewrap|stk|1.16.3
even-better-toml|tam|0.19.2
pdf|tom|1.2.2
simple-rst|tro|1.5.4
vscode-ltex|val|13.1.0
vscode-icons|vsc|12.6.0
vim|vsc|1.26.2
txt-syntax|xsh|0.2.2
markdown-all-in-one|yzh|3.5.1

</details>

**Comment by sztal** [comment]
Hi there, any progress on this issue?

**Comment by StellaHuang95** [comment]
Hi @sztal, I will try to repro it. Meanwhile, could you provide more loggings as described in [the troubleshooting guide](https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md#filing-an-issue)?

**Comment by sztal** [comment]
Hi, 

here is the Pylance output.

```
2023-11-13 23:38:15.846 [info] [Info  - 23:38:15] (32247) Pylance language server 2023.11.10 (pyright 088ebaa5) starting
2023-11-13 23:38:15.847 [info] [Info  - 23:38:15] (32247) Server root directory: /home/sztal/.vscode/extensions/ms-python.vscode-pylance-2023.11.10/dist
2023-11-13 23:38:15.847 [info] [Info  - 23:38:15] (32247) Starting service instance "GuestXR-Envir-Game"
2023-11-13 23:38:15.887 [info] (32247) No configuration file found.
2023-11-13 23:38:15.888 [info] (32247) No pyproject.toml file found.
2023-11-13 23:38:15.888 [info] [Info  - 23:38:15] (32247) Setting pythonPath for service "GuestXR-Envir-Game": "/home/sztal/miniconda3/bin/python"
2023-11-13 23:38:15.888 [info] [Info  - 23:38:15] (32247) Setting environmentName for service "GuestXR-Envir-Game": "3.11.5 (base conda)"
2023-11-13 23:38:15.924 [info] [Info  - 23:38:15] (32247) Assuming Python version 3.11
2023-11-13 23:38:15.925 [info] (32247) Assuming Python platform Linux
2023-11-13 23:38:15.985 [info] [Info  - 23:38:15] (32247) Search paths for /home/sztal/Dropbox/Akademia/ISS/Projekty/GuestXR/GuestXR-Envir-Game
2023-11-13 23:38:15.986 [info] [Info  - 23:38:15] (32247)   /home/sztal/.vscode/extensions/ms-python.vscode-pylance-2023.11.10/dist/typeshed-fallback/stdlib
2023-11-13 23:38:15.986 [info] [Info  - 23:38:15] (32247)   /home/sztal/Dropbox/Akademia/ISS/Projekty/GuestXR/GuestXR-Envir-Game
2023-11-13 23:38:15.986 [info] [Info  - 23:38:15] (32247)   /home/sztal/Dropbox/Akademia/ISS/Projekty/GuestXR/GuestXR-Envir-Game/typings
2023-11-13 23:38:15.986 [info] [Info  - 23:38:15] (32247)   /home/sztal/.vscode/extensions/ms-python.vscode-pylance-2023.11.10/dist/typeshed-fallback/stubs/...
2023-11-13 23:38:15.987 [info] [Info  - 23:38:15] (32247)   /home/sztal/.vscode/extensions/ms-python.vscode-pylance-2023.11.10/dist/bundled/stubs
2023-11-13 23:38:15.987 [info] [Info  - 23:38:15] (32247)   /home/sztal/miniconda3/lib/python3.11
2023-11-13 23:38:15.987 [info] [Info  - 23:38:15] (32247)   /home/sztal/miniconda3/lib/python3.11/lib-dynload
2023-11-13 23:38:15.987 [info] [Info  - 23:38:15] (32247)   /home/sztal/miniconda3/lib/python3.11/site-packages
2023-11-13 23:38:15.987 [info] [Info  - 23:38:15] (32247) Adding fs watcher for library directories:
 /home/sztal/miniconda3/lib/python3.11
/home/sztal/miniconda3/lib/python3.11/lib-dynload
/home/sztal/miniconda3/lib/python3.11/site-packages
2023-11-13 23:38:15.988 [info] [Info  - 23:38:15] (32247) Adding fs watcher for directories:
 /home/sztal/Dropbox/Akademia/ISS/Projekty/GuestXR/GuestXR-Envir-Game
2023-11-13 23:38:15.988 [info] (32247) Searching for source files
2023-11-13 23:38:15.991 [info] [Info  - 23:38:15] (32247) Found 1 source file
2023-11-13 23:38:15.992 [info] (32247) pytest configurations: {"message":"script","classes":["Test"],"files":["test_*.py","*_test.py"],"functions":["test"]}
2023-11-13 23:38:15.992 [info] (32247) pytest configurations: {"message":"script","classes":["Test"],"files":["test_*.py","*_test.py"],"functions":["test"]}
2023-11-13 23:38:16.458 [info] (32247) pytest configurations: {"message":"request cancelled","classes":["Test"],"files":["test_*.py","*_test.py"],"functions":["test"]}
2023-11-13 23:38:16.508 [info] (32247) [BG(1)] indexing: /home/sztal/Dropbox/Akademia/ISS/Projekty/GuestXR/GuestXR-Envir-Game/model.py ...
2023-11-13 23:38:16.508 [info] (32247) [BG(1)]   parsing: /home/sztal/Dropbox/Akademia/ISS/Projekty/GuestXR/GuestXR-Envir-Game/model.py [fs read 1ms] (48ms)
2023-11-13 23:38:16.570 [info] (32247) [BG(1)]   parsing: /home/sztal/.vscode/extensions/ms-python.vscode-pylance-2023.11.10/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 1ms] (61ms)
2023-11-13 23:38:16.621 [info] (32247) [BG(1)]   binding: /home/sztal/.vscode/extensions/ms-python.vscode-pylance-2023.11.10/dist/typeshed-fallback/stdlib/builtins.pyi (49ms)
2023-11-13 23:38:16.624 [info] (32247) [BG(1)]   binding: /home/sztal/Dropbox/Akademia/ISS/Projekty/GuestXR/GuestXR-Envir-Game/model.py (5ms)
2023-11-13 23:38:16.627 [info] (32247) [BG(1)] indexing: /home/sztal/Dropbox/Akademia/ISS/Projekty/GuestXR/GuestXR-Envir-Game/model.py [found 2] (168ms)
2023-11-13 23:38:16.699 [info] (32247) pytest configurations: {"message":"script","classes":["Test"],"files":["test_*.py","*_test.py"],"functions":["test"]}
```

And since I am using `conda` here is the content of `environment.yml`:

```
name: base
channels:
  - defaults
  - conda-forge
dependencies:
  - _libgcc_mutex=0.1=conda_forge
  - _openmp_mutex=4.5=2_kmp_llvm
  - aiofiles=22.1.0=py311h06a4308_0
  - aiosqlite=0.18.0=py311h06a4308_0
  - anyio=3.5.0=py311h06a4308_0
  - argon2-cffi=21.3.0=pyhd3eb1b0_0
  - argon2-cffi-bindings=21.2.0=py311h5eee18b_0
  - arpack=3.7.0=hdefa2d7_2
  - arrow=1.2.3=py311h06a4308_1
  - astroid=2.14.2=py311h06a4308_0
  - asttokens=2.0.5=pyhd3eb1b0_0
  - attrs=23.1.0=py311h06a4308_0
  - babel=2.11.0=py311h06a4308_0
  - backcall=0.2.0=pyhd3eb1b0_0
  - beautifulsoup4=4.12.2=py311h06a4308_0
  - binaryornot=0.4.4=pyhd3eb1b0_1
  - blas=2.116=openblas
  - blas-devel=3.9.0=16_linux64_openblas
  - bleach=4.1.0=pyhd3eb1b0_0
  - boltons=23.0.0=py311h06a4308_0
  - bottleneck=1.3.5=py311hbed6279_0
  - bounded-pool-executor=0.0.3=pyhd8ed1ab_0
  - brotli=1.0.9=h5eee18b_7
  - brotli-bin=1.0.9=h5eee18b_7
  - brotli-python=1.0.9=py311h6a678d5_7
  - bzip2=1.0.8=h7b6447c_0
  - c-ares=1.19.1=h5eee18b_0
  - ca-certificates=2023.08.22=h06a4308_0
  - certifi=2023.7.22=py311h06a4308_0
  - cffi=1.15.1=py311h5eee18b_3
  - chardet=4.0.0=py311h06a4308_1003
  - charset-normalizer=2.0.4=pyhd3eb1b0_0
  - click=8.1.7=py311h06a4308_0
  - colorama=0.4.6=py311h06a4308_0
  - comm=0.1.2=py311h06a4308_0
  - conda=23.9.0=py311h06a4308_0
  - conda-content-trust=0.2.0=py311h06a4308_0
  - conda-libmamba-solver=23.9.3=py311h06a4308_0
  - conda-package-handling=2.2.0=py311h06a4308_0
  - conda-package-streaming=0.9.0=py311h06a4308_0
  - contourpy=1.0.5=py311hdb19cb5_0
  - cookiecutter=1.7.3=pyhd3eb1b0_0
  - cryptography=41.0.3=py311hdda0065_0
  - cycler=0.11.0=pyhd3eb1b0_0
  - cyrus-sasl=2.1.28=h52b45da_1
  - cython=3.0.0=py311h5eee18b_0
  - dbus=1.13.18=hb2f20db_0
  - debugpy=1.6.7=py311h6a678d5_0
  - decorator=5.1.1=pyhd3eb1b0_0
  - deepdiff=6.6.1=pyhd8ed1ab_0
  - defusedxml=0.7.1=pyhd3eb1b0_0
  - dill=0.3.7=py311h06a4308_0
  - entrypoints=0.4=py311h06a4308_0
  - et_xmlfile=1.1.0=py311h06a4308_0
  - executing=0.8.3=pyhd3eb1b0_0
  - expat=2.5.0=h6a678d5_0
  - flake8=6.0.0=py311h06a4308_0
  - fmt=9.1.0=hdb19cb5_0
  - fontconfig=2.14.1=h4c34cd2_2
  - fonttools=4.25.0=pyhd3eb1b0_0
  - freetype=2.12.1=h4a9f257_0
  - giflib=5.2.1=h5eee18b_3
  - glib=2.69.1=he621ea3_2
  - glpk=5.0=h445213a_0
  - gmp=6.2.1=h295c915_3
  - gst-plugins-base=1.14.1=h6a678d5_1
  - gstreamer=1.14.1=h5eee18b_1
  - hdbscan=0.8.33=py311h1f0f07a_4
  - icu=58.2=he6710b0_3
  - idna=3.4=py311h06a4308_0
  - iniconfig=1.1.1=pyhd3eb1b0_0
  - ipdb=0.13.13=pyhd8ed1ab_0
  - ipykernel=6.25.0=py311h92b7b1e_0
  - ipython=8.15.0=py311h06a4308_0
  - ipython_genutils=0.2.0=pyhd3eb1b0_1
  - ipywidgets=8.0.4=py311h06a4308_0
  - isort=5.9.3=pyhd3eb1b0_0
  - jedi=0.18.1=py311h06a4308_1
  - jinja2=3.1.2=py311h06a4308_0
  - jinja2-time=0.2.0=pyhd3eb1b0_3
  - joblib=1.2.0=py311h06a4308_0
  - jpeg=9e=h5eee18b_1
  - json5=0.9.6=pyhd3eb1b0_0
  - jsonpatch=1.32=pyhd3eb1b0_0
  - jsonpointer=2.1=pyhd3eb1b0_0
  - jsonschema=4.17.3=py311h06a4308_0
  - jupyter_client=7.4.9=py311h06a4308_0
  - jupyter_core=5.3.0=py311h06a4308_0
  - jupyter_events=0.6.3=py311h06a4308_0
  - jupyter_server=1.23.4=py311h06a4308_0
  - jupyter_server_fileid=0.9.0=py311h06a4308_0
  - jupyter_server_ydoc=0.8.0=py311h06a4308_1
  - jupyter_ydoc=0.2.4=py311h06a4308_0
  - jupyterlab=3.6.3=py311h06a4308_0
  - jupyterlab_pygments=0.1.2=py_0
  - jupyterlab_server=2.22.0=py311h06a4308_0
  - jupyterlab_widgets=3.0.5=py311h06a4308_0
  - kiwisolver=1.4.4=py311h6a678d5_0
  - krb5=1.20.1=h143b758_1
  - lazy-object-proxy=1.6.0=py311h5eee18b_0
  - lcms2=2.12=h3be6417_0
  - ld_impl_linux-64=2.38=h1181459_1
  - lerc=3.0=h295c915_0
  - libarchive=3.6.2=h6ac8c49_2
  - libblas=3.9.0=16_linux64_openblas
  - libbrotlicommon=1.0.9=h5eee18b_7
  - libbrotlidec=1.0.9=h5eee18b_7
  - libbrotlienc=1.0.9=h5eee18b_7
  - libcblas=3.9.0=16_linux64_openblas
  - libclang=14.0.6=default_hc6dbbc7_1
  - libclang13=14.0.6=default_he11475f_1
  - libcups=2.4.2=h2d74bed_1
  - libcurl=8.4.0=h251f7ec_0
  - libdeflate=1.17=h5eee18b_1
  - libedit=3.1.20221030=h5eee18b_0
  - libev=4.33=h7f8727e_1
  - libevent=2.1.12=hdbd6064_1
  - libffi=3.4.4=h6a678d5_0
  - libgcc-ng=13.2.0=h807b86a_2
  - libgfortran-ng=11.2.0=h00389a5_1
  - libgfortran5=11.2.0=h1234567_1
  - liblapack=3.9.0=16_linux64_openblas
  - liblapacke=3.9.0=16_linux64_openblas
  - libllvm14=14.0.6=hdb19cb5_3
  - libmamba=1.5.1=haf1ee3a_0
  - libmambapy=1.5.1=py311h2dafd23_0
  - libnghttp2=1.57.0=h2d74bed_0
  - libopenblas=0.3.21=pthreads_h78a6416_3
  - libpng=1.6.39=h5eee18b_0
  - libpq=12.15=hdbd6064_1
  - libsodium=1.0.18=h7b6447c_0
  - libsolv=0.7.24=he621ea3_0
  - libssh2=1.10.0=hdbd6064_2
  - libstdcxx-ng=13.2.0=h7e041cc_2
  - libtiff=4.5.1=h6a678d5_0
  - libuuid=1.41.5=h5eee18b_0
  - libwebp=1.3.2=h11a3e52_0
  - libwebp-base=1.3.2=h5eee18b_0
  - libxcb=1.15=h7f8727e_0
  - libxkbcommon=1.0.1=h5eee18b_1
  - libxml2=2.10.4=hcbfbd50_0
  - libxslt=1.1.37=h2085143_0
  - libzlib=1.2.13=hd590300_5
  - llvm-openmp=14.0.6=h9e868ea_0
  - llvmlite=0.40.0=py311he621ea3_0
  - lxml=4.9.3=py311hdbbb534_0
  - lz4-c=1.9.4=h6a678d5_0
  - markupsafe=2.1.1=py311h5eee18b_0
  - matplotlib=3.7.2=py311h06a4308_0
  - matplotlib-base=3.7.2=py311ha02d727_0
  - matplotlib-inline=0.1.6=py311h06a4308_0
  - mccabe=0.7.0=pyhd3eb1b0_0
  - metis=5.1.0=hf484d3e_4
  - mistune=0.8.4=py311h5eee18b_1000
  - mpfr=4.0.2=hb69a4c5_1
  - munkres=1.1.4=py_0
  - mypy=1.5.1=py311h06a4308_0
  - mypy_extensions=1.0.0=py311h06a4308_0
  - mysql=5.7.24=h721c034_2
  - nbclassic=0.5.5=py311h06a4308_0
  - nbclient=0.5.13=py311h06a4308_0
  - nbconvert=6.5.4=py311h06a4308_0
  - nbformat=5.9.2=py311h06a4308_0
  - ncurses=6.4=h6a678d5_0
  - nest-asyncio=1.5.6=py311h06a4308_0
  - notebook=6.5.4=py311h06a4308_1
  - notebook-shim=0.2.2=py311h06a4308_0
  - nspr=4.35=h6a678d5_0
  - nss=3.89.1=h6a678d5_0
  - numba=0.57.1=py311h96b013e_0
  - numexpr=2.8.7=py311h812550d_0
  - numpy=1.24.3=py311h434b4ae_0
  - numpy-base=1.24.3=py311h4367d22_0
  - openblas=0.3.21=pthreads_h320a7e8_3
  - openjpeg=2.4.0=h3ad879b_0
  - openpyxl=3.0.10=py311h5eee18b_0
  - openssl=3.0.12=h7f8727e_0
  - ordered-set=4.1.0=py311h06a4308_0
  - orjson=3.8.8=py311h52d8a92_0
  - packaging=23.1=py311h06a4308_0
  - pandas=2.1.1=py311ha02d727_0
  - pandas-stubs=1.5.3.230203=py311h06a4308_0
  - pandocfilters=1.5.0=pyhd3eb1b0_0
  - parso=0.8.3=pyhd3eb1b0_0
  - pcre=8.45=h295c915_0
  - pcre2=10.42=hebb0a14_0
  - pexpect=4.8.0=pyhd3eb1b0_3
  - pickleshare=0.7.5=pyhd3eb1b0_1003
  - pillow=10.0.1=py311ha6cbd5a_0
  - pip=23.3=py311h06a4308_0
  - platformdirs=3.10.0=py311h06a4308_0
  - pluggy=1.0.0=py311h06a4308_1
  - ply=3.11=py311h06a4308_0
  - poyo=0.5.0=pyhd3eb1b0_0
  - pqdm=0.2.0=pyhd8ed1ab_0
  - prometheus_client=0.14.1=py311h06a4308_0
  - prompt-toolkit=3.0.36=py311h06a4308_0
  - psutil=5.9.0=py311h5eee18b_0
  - ptyprocess=0.7.0=pyhd3eb1b0_2
  - pure_eval=0.2.2=pyhd3eb1b0_0
  - pybind11-abi=4=hd3eb1b0_1
  - pycodestyle=2.10.0=py311h06a4308_0
  - pycosat=0.6.6=py311h5eee18b_0
  - pycparser=2.21=pyhd3eb1b0_0
  - pyflakes=3.0.1=py311h06a4308_0
  - pygments=2.15.1=py311h06a4308_1
  - pylint=2.16.2=py311h06a4308_0
  - pyopenssl=23.2.0=py311h06a4308_0
  - pyparsing=3.0.9=py311h06a4308_0
  - pyqt=5.15.7=py311h6a678d5_0
  - pyqt5-sip=12.11.0=py311h6a678d5_0
  - pyrsistent=0.18.0=py311h5eee18b_0
  - pysocks=1.7.1=py311h06a4308_0
  - pytest=7.4.0=py311h06a4308_0
  - python=3.11.5=h955ad1f_0
  - python-dateutil=2.8.2=pyhd3eb1b0_0
  - python-fastjsonschema=2.16.2=py311h06a4308_0
  - python-igraph=0.10.4=py311hcb6f00c_0
  - python-json-logger=2.0.7=py311h06a4308_0
  - python-slugify=5.0.2=pyhd3eb1b0_0
  - python-tzdata=2023.3=pyhd3eb1b0_0
  - python_abi=3.11=2_cp311
  - pytz=2023.3.post1=py311h06a4308_0
  - pyyaml=6.0.1=py311h5eee18b_0
  - pyzmq=23.2.0=py311h6a678d5_0
  - qt-main=5.15.2=h7358343_9
  - qt-webengine=5.15.9=h9ab4d14_7
  - qtwebkit=5.212=h3fafdc1_5
  - readline=8.2=h5eee18b_0
  - reproc=14.2.4=h295c915_1
  - reproc-cpp=14.2.4=h295c915_1
  - requests=2.31.0=py311h06a4308_0
  - rfc3339-validator=0.1.4=py311h06a4308_0
  - rfc3986-validator=0.1.1=py311h06a4308_0
  - ruamel.yaml=0.17.21=py311h5eee18b_0
  - scikit-learn=1.3.0=py311ha02d727_0
  - scipy=1.11.3=py311h24aa872_0
  - seaborn=0.12.2=py311h06a4308_0
  - send2trash=1.8.0=pyhd3eb1b0_1
  - setuptools=68.0.0=py311h06a4308_0
  - sip=6.6.2=py311h6a678d5_0
  - six=1.16.0=pyhd3eb1b0_1
  - sniffio=1.2.0=py311h06a4308_1
  - soupsieve=2.5=py311h06a4308_0
  - sqlite=3.41.2=h5eee18b_0
  - stack_data=0.2.0=pyhd3eb1b0_0
  - suitesparse=5.10.1=he2db622_2
  - tbb=2021.8.0=hdb19cb5_0
  - terminado=0.17.1=py311h06a4308_0
  - text-unidecode=1.3=pyhd3eb1b0_0
  - texttable=1.6.4=pyhd3eb1b0_0
  - threadpoolctl=2.2.0=pyh0d69192_0
  - tinycss2=1.2.1=py311h06a4308_0
  - tk=8.6.12=h1ccaba5_0
  - toml=0.10.2=pyhd3eb1b0_0
  - tomlkit=0.11.1=py311h06a4308_0
  - tornado=6.3.3=py311h5eee18b_0
  - tqdm=4.65.0=py311h92b7b1e_0
  - traitlets=5.7.1=py311h06a4308_0
  - truststore=0.8.0=py311h06a4308_0
  - types-pytz=2022.4.0.0=py311h06a4308_1
  - typing-extensions=4.7.1=py311h06a4308_0
  - typing_extensions=4.7.1=py311h06a4308_0
  - tzdata=2023c=h04d1e81_0
  - unidecode=1.2.0=pyhd3eb1b0_0
  - urllib3=1.26.18=py311h06a4308_0
  - wcwidth=0.2.5=pyhd3eb1b0_0
  - webencodings=0.5.1=py311h06a4308_1
  - websocket-client=0.58.0=py311h06a4308_4
  - wheel=0.41.2=py311h06a4308_0
  - widgetsnbextension=4.0.5=py311h06a4308_0
  - wrapt=1.14.1=py311h5eee18b_0
  - xz=5.4.2=h5eee18b_0
  - y-py=0.5.9=py311h52d8a92_0
  - yaml=0.2.5=h7b6447c_0
  - yaml-cpp=0.7.0=h295c915_1
  - ypy-websocket=0.8.2=py311h06a4308_0
  - zeromq=4.3.4=h2531618_0
  - zlib=1.2.13=hd590300_5
  - zstandard=0.19.0=py311h5eee18b_0
  - zstd=1.5.5=hc292b87_0
  - pip:
      - alabaster==0.7.13
      - annotated-types==0.5.0
      - argcomplete==3.1.2
      - blis==0.7.11
      - catalogue==2.0.10
      - cmake==3.25.0
      - confection==0.1.3
      - cymem==2.0.8
      - docutils==0.18.1
      - filelock==3.9.0
      - fn-py==0.6.0
      - fsspec==2023.9.2
      - huggingface-hub==0.17.3
      - imagesize==1.4.1
      - langcodes==3.3.0
      - latexcodec==2.0.1
      - lit==15.0.7
      - mini-lambda==2.2.3
      - more-itertools==10.1.0
      - mpmath==1.2.1
      - murmurhash==1.0.10
      - networkx==3.0
      - pathy==0.10.2
      - pipx==1.2.0
      - preshed==3.0.9
      - pybtex==0.24.0
      - pybtex-docutils==1.0.3
      - pydantic==1.10.13
      - pydantic-core==2.10.1
      - regex==2023.8.8
      - smart-open==6.4.0
      - snowballstemmer==2.2.0
      - sphinx==7.2.6
      - sphinx-rtd-theme==1.3.0
      - sphinxcontrib-applehelp==1.0.7
      - sphinxcontrib-bibtex==2.5.0
      - sphinxcontrib-devhelp==1.0.5
      - sphinxcontrib-htmlhelp==2.0.4
      - sphinxcontrib-jquery==4.1
      - sphinxcontrib-jsmath==1.0.1
      - sphinxcontrib-qthelp==1.0.6
      - sphinxcontrib-serializinghtml==1.1.9
      - srsly==2.4.8
      - sympy==1.11.1
      - thinc==8.1.12
      - tokenizers==0.13.3
      - torch==2.0.1+cu118
      - torchaudio==2.0.2+cu118
      - torchvision==0.15.2+cu118
      - transformers==4.25.1
      - triton==2.0.0
      - typer==0.7.0
      - userpath==1.9.1
      - wasabi==0.10.1
prefix: /home/sztal/miniconda3
```

System information are provided in the initial post.

**Comment by heejaechang** [comment]
we watch all python search paths to make sure we detect when new packages are installed/uninstalled or modified. and we also watch user files under workspace. but that said, by watching, we ask vscode to watch them. we don't do it by ourselves.

**Comment by sztal** [comment]
Ok, I see. Actually setting watchers to the OS-allowed maximum solves the problem in most cases. That is, I do not get the warning anymore. Thanks!

**Comment by yellowtailfan** [comment]
I've been getting the "Unable to watch for file changes" warning from VS Code for many months, and tried several times to diagnose and fix it. Thanks to this bug report I finally tracked it down.

I had the top folder of my local checkouts tree in my `PYTHONPATH`, which has 4.3m files inside 3.1m folders. So VS Code was consuming my entire custom 2m limit on inotify `max_user_watches`. I measured this using the excellent [inotify-info](https://github.com/mikesart/inotify-info) tool.

I actually need to import some modules from the top level of my checkouts tree, but to work around the problem I made a sub-folder with symlinks to the specific module folders that I want to be able to import, added the sub-folder to my `PYTHONPATH` and removed the top-level checkouts folder from my `PYTHONPATH`. Now my VS code watcher count is only 382!

It took me a while to track this down so I thought I would share my story here in case it helps others. It would also be helpful to add "check your PYTHONPATH" to the [VS Code FAQ on this issue](https://code.visualstudio.com/docs/setup/linux#_visual-studio-code-is-unable-to-watch-for-file-changes-in-this-large-workspace-error-enospc).
